scripts/evaluate_release_tag.py

```diff
@@ -51,6 +55,11 @@ def should_build_docs(tag: Version) -> bool:
     latest_version = non_alpha_releases[-1]
     print(f"The latest non-alpha/rc/nightly tag is {latest_version}.")
 
+    non_beta_releases = filter_non_beta_relases(non_alpha_releases)
```

Comment:
Should we actually consolidate `filter_non_alpha_releases` and `filter_non_beta_releases` to a function that filters for GA releases? rather than have two functions with similar end goal/use case?

Reply:
> if we want to allow builds on the beta versions like 3.7.0b3

Not sure either, I think sometimes betas come with docs changes and they should be indeed build, @m-vdb do you agree?

> if the phrasing of non-alpha/rc/nightly is intentional to not include beta 

I think they just meant fetch the latest GA version (before we weren't doing that many betas really), but maybe wait for @sanchariGr to confirm? I think she's made changes to this script before.

Reply:
I think we can bucket releases into 2 categories:
- pre-releases: nightly, alpha, beta, rcs
- general availability releases: micros, minor, major

Reply:
Got it, I've simplified the function now with some cleanup